lib/hca/service.rb

```diff
@@ -39,24 +37,6 @@ def health_check
 
     private
 
-    def handle_timeout(e)
-      StatsD.increment(TIMEOUT_KEY)
-      log_exception_to_sentry(e, {}, {}, 'warning')
-      raise Common::Exceptions::GatewayTimeout
-    end
-
-    def post_submission(submission)
-      perform(:post, '', submission.body)
-    rescue Common::Exceptions::GatewayTimeout => e
-      handle_timeout(e)
-    rescue Common::Client::Errors::HTTPError => e
-      if e.code.to_s == '503'
-        handle_timeout(e)
```

Comment:
it looks like these won't be considered timeouts anymore?

Reply:
Good catch.  I assumed that the middleware would effectively make this unnecessary, but I now realize that explicit 503 responses would get through.  I'll write up a corresponding response middleware for these.